src/hooks/useRewardShare.ts

```diff
@@ -0,0 +1,54 @@
+import { BigNumber, ethers } from 'ethers';
+import { useWeb3React } from '@web3-react/core';
+import { useEffect, useState } from 'react';
+import { useRewardPrograms } from 'store/rewardPrograms';
+import { useCrucibles } from 'store/crucibles';
+import bigNumberishToNumber from 'utils/bigNumberishToNumber';
+import { useRewardProgramTokens } from 'store/rewardProgramTokens';
+
+export function useRewardShare(crucibleAddress: string) {
+  const { cruciblesLoading } = useCrucibles();
+  const [rewardsShare, setRewardsShare] = useState<number>(0);
+  const { library } = useWeb3React();
+  const signer = library.getSigner();
+  const { currentRewardProgram } = useRewardPrograms();
+  const { programTokensLoading } = useRewardProgramTokens();
+
+  const getRewardShare = async () => {
+    try {
+      // create the contract instances for fetching subscription data
+      const rewardProgramContract = new ethers.Contract(
+        currentRewardProgram.address,
+        currentRewardProgram.abi,
+        signer
+      );
+
+      // Get reward share
+      const currentVaultStakeUnits: BigNumber =
+        await rewardProgramContract.getCurrentVaultStakeUnits(crucibleAddress);
+      const currentVaultStakeUnitsNumber = bigNumberishToNumber(
+        currentVaultStakeUnits
+      );
+      const totalStakeUnits =
+        await rewardProgramContract.getCurrentTotalStakeUnits();
+      const currentTotalStakeUnitsNumber =
+        bigNumberishToNumber(totalStakeUnits);
+      const rewardsShare =
+        currentVaultStakeUnitsNumber / currentTotalStakeUnitsNumber;
+      setRewardsShare(rewardsShare);
```

Comment:
Lets get rid of using bigNumberishToNumber for calcs, sorry it was my implementation but with hindsight I would have avoided it if I knew it was problematic.

```suggestion
      const currentVaultStakeUnits: BigNumber =
        await rewardProgramContract.getCurrentVaultStakeUnits(crucibleAddress);
       
      const totalStakeUnits = await rewardProgramContract.getCurrentTotalStakeUnits();
        
      const rewardsShare = currentVaultStakeUnits.div(totalStakeUnits);
      setRewardsShare(rewardsShare);
```

Reply:
![Screen Shot 2021-07-07 at 9 41 55 PM](https://user-images.githubusercontent.com/15269424/124863580-2aa74780-df6c-11eb-95cd-eaf2a336c1d3.png)


Hey @joehawary I've been debugging this for 45min + and can't get the same values from the BigNumber approach. Not sure what the issue is but I am getting `0` every time.

Starting to think it's not worth more time to figure out, but I'm also not aware of the issues you were seeing with exporting this value as a `number`?

Reply:
<img width="680" alt="Screen Shot 2021-07-07 at 9 45 18 PM" src="https://user-images.githubusercontent.com/15269424/124863884-c2a53100-df6c-11eb-9e25-dfea5827d7ca.png">